app/timetables/models.py

```diff
@@ -72,12 +72,21 @@ class Timetable(TimestampMixin):
     name = models.CharField(max_length=255, unique=True)
     code = models.CharField(max_length=60, unique=True)
     api_key = models.CharField(max_length=255, unique=True)
-    cycle_length = models.IntegerField()
-    current_cycle_day = models.IntegerField()
+    cycle_length = models.PositiveSmallIntegerField(
+        validators=[MinValueValidator(1)]
+    )
+    current_cycle_day = models.PositiveSmallIntegerField(
+        validators=[MinValueValidator(1)]
+    )
 
     def clean(self):
-        if self.current_cycle_day > self.cycle_length:
-            raise ValidationError(_('current_cycle_day must not be greater than cycle_length.'))
+        # Ensure current_cycle_day and cycle_length are not None before compare
+        if self.current_cycle_day and self.cycle_length:
+            if self.current_cycle_day > self.cycle_length:
```

Comment:
I had that check to ensure that there are no None type values being compared. Without that, we would have this error 
![image](https://cloud.githubusercontent.com/assets/10073270/18272051/802dbb1c-742d-11e6-9b6a-60ba8881624b.png)